### PR TITLE
Twitter popups

### PR DESCRIPTION
**Pull request by alexsroth**

### CODE REVIEW

File: 01-builder-visualization/exercises/twitter-dogs.md [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+<hr>
+
+## 2. Style <a name="sty"></a>
+
+### 2. 1. Styling Time-Based Elements
+
+* After clicking on **Dogs of Twitter** layer, it will direct you to the `STYLE` tab.
+* Change the **aggregation style** to `Animated`
+* Set the style options as follows:
+  * **FILL**:click on the color, click on `BY VALUE`, style by `category_terms`. Color these as you see fit by clicking on the color in line with the term you want to change.
```
Comment (michellechandra) [inline 01-builder-visualization/exercises/twitter-dogs.md]: - space needed between Fill: and "click"

- be more specific: "click on the orange color bar, in the color wheel that opens, select "'BY VALUE"'

File: 01-builder-visualization/exercises/twitter-dogs.md [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+  * **COLUMN**: `postedtime`
+  * **DURATION**: `30`
```
Comment (michellechandra) [inline 01-builder-visualization/exercises/twitter-dogs.md]: Prob don't need to specify these options (duration onward) since they are the defaults and the user isn't changing them for the tutorial.

**Comment by jsanz** [comment]
Another thing, for some reason there are some 1-pixel lines on the tiles of your screenshots @alexsroth, it would be nice to repeat them without them. Maybe there was some zoom applied in your browser?

File: 01-builder-visualization/exercises/twitter-dogs.md [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+  * **TRAILS**: `2`
+  * **RESOLUTION**: `4`
+
+![animated settings](imgs/01-dog-01.png)
+
+* Click on the `LEGEND` tab. Set the style to `CUSTOM`. Check the box to add title and enter `By breed` to give your legend enough detail to be useful.
+
+### 2. 2. Styling Pop-ups
+
+* In order to add pop-ups to our map, we will need to add the `twitter_dogs` layer again. In the top left corner, press the back arrow to navigate back to the `LAYERS` menu.
```
Comment (michellechandra) [inline 01-builder-visualization/exercises/twitter-dogs.md]: I would add some additional context here as to why we have to add the popups to a second layer, something like:

Pop ups can't be added to animated map layer, so will need to add our Twitter layer again as a static data layer to enable pop ups on that layer.

File: 01-builder-visualization/exercises/twitter-dogs.md [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+  * You will notice that your time slider is already present.
+* Click `ADD`.
+  * In the options of the **CATEGORY** tab, check the *category_terms* widget. Make sure you are adding a widget from the `A0` layer node.
+  * In the options of the **FORMULA** tab, check `retweetcount` and `actor_followerscount`, again making sure you are adding the widget with the source as `A0`.
+* Click on `CONTINUE`.
+
+### 3. 2. Order and edit widgets
+
+* Navigate back to the **WIDGETS** pane.
+* **Click the *time widget*.**
```
Comment (michellechandra) [inline 01-builder-visualization/exercises/twitter-dogs.md]: Mention they need to double click on the title at the top of the widget pane to edit/rename.

File: 01-builder-visualization/exercises/twitter-dogs.md [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+  - Change the title to **When are people tweeting?**.
+  - You can change the color of the widget bars now too if you'd like.
```
Comment (michellechandra) [inline 01-builder-visualization/exercises/twitter-dogs.md]: Be more specific here - something like - You can also change the color of the widget timeseries bars by selecting the FILL color option at the bottom of the widget pane.

File: 01-builder-visualization/exercises/twitter-dogs.md [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+
+* **Click the *retweetcount* widget.**
+  - Change the title to **How popular are these tweets?**
+  - Make sure the type is set to `FORMULA`.
+  - Change the **OPERATION** to `SUM(retweetcount)`
+  - Add the **SUFFIX**: ` retweets` to let your viewers know what the number they are looking at means.
```
Comment (michellechandra) [inline 01-builder-visualization/exercises/twitter-dogs.md]: -suffix - specify that they should add a space before "retweets" so its 8,029 retweets rather than 8209retweets

File: 01-builder-visualization/exercises/twitter-dogs.md [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+  - Widgets are only as good as the data you provide to them. You don't always want the viewers to interpret what each item means. Go ahead and add a description for the widget. In this case, I am going to explain the purpose of a retweet. Here's what I am saying:
```
Comment (michellechandra) [inline 01-builder-visualization/exercises/twitter-dogs.md]: Add some more specifics here in case they don't understand how, something like - From the widgets pane, you can reorder the widgets and how they are displayed on the map by dragging and dropping each widget option in the panel to reorder.

File: 01-builder-visualization/exercises/twitter-dogs.md [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+
+* Reorder the position of widgets as you see fit.
+
+![widget-options](imgs/01-dog-06.png)
+
+<hr>
+
+## 4. Publish and share map <a name="pub"></a>
+
+* At the bottom of the main menu, click on **SHARE** at the right button corner.
```
Comment (michellechandra) [inline 01-builder-visualization/exercises/twitter-dogs.md]: Instead of "main menu", say layer pane.

File: 01-builder-visualization/exercises/twitter-dogs.md [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+* Go to **PUBLISH** tab and click on **PUBLISH** button that is below the Map name in order to share our map.
```
Comment (michellechandra) [inline 01-builder-visualization/exercises/twitter-dogs.md]: They would first need to press publish to publish the map, then select the publish tab to see the share options...

File: 01-builder-visualization/exercises/twitter-dogs.md [hunk 10]
```diff
@@ -0,0 +1,199 @@
...
+* After clicking on **PUBLISH**, we can select the options that we want to share our map.
```
Comment (michellechandra) [inline 01-builder-visualization/exercises/twitter-dogs.md]: I would reword this - something like - After publishing, we can see from the Publish tab different options for sharing the dataset including a URL to a public map page and iframe embed code for sharing the map on an external website.

**Comment by michellechandra** [comment]
Christa has a document on official Builder terminology you may want to reference, I would reach out to her for that doc!

File: 01-builder-visualization/exercises/twitter-dogs.md [hunk 11]
```diff
@@ -0,0 +1,202 @@
...
+
+* *Degree of Difficulty*: :star::star::star:
+* *Goal*: Create a time-enabled visualization that includes useful pop-ups using data pulled from Twitter with BUILDER.
+* *Features Highlighted*:
+  * Style:
+    - Enabling time-based elements of your data for advanced visualization;
+    - Making the most of your data by adding and altering pop-ups to display useful information at a glance.
+  * Layers: how to work with layer nodes
+  * Widgets: Formulas + Histograms
+* *Datasests needed*:
```
Comment (michellechandra) [inline 01-builder-visualization/exercises/twitter-dogs.md]: Typo - datasets not datasests

File: 01-builder-visualization/exercises/twitter-dogs.md [hunk 12]
```diff
@@ -0,0 +1,202 @@
...
+  * `twitter_dogs` dataset: twitter data on based on dog breed search terms from 6/11/2017 - 7/10/17.
```
Comment (michellechandra) [inline 01-builder-visualization/exercises/twitter-dogs.md]: Typo - extra "on"

File: 01-builder-visualization/exercises/twitter-dogs.md [hunk 13]
```diff
@@ -0,0 +1,202 @@
...
+### 1. 1. Import `twitter_dogs` file into your DATASETS dashboard.
+
+* Download the dataset from [here](https://builder-demo.carto.com/api/v2/sql?q=SELECT+*+FROM+twitter_dogs&format=csv&filename=twitter_dogs).
+* Go to your account and import it using the **NEW DATASET** options.
+* Select **CONNECT DATASET**.
+* Drag and drop the file.
+
+### 1. 2. Dataset view
+
+* Take a look on the dataset.
```
Comment (michellechandra) [inline 01-builder-visualization/exercises/twitter-dogs.md]: prob "on" should be "at", take a look on the dataset sounds awkward grammatically

**Comment by michellechandra** [comment]
Looks good!
